File: 2655-rearrange-array-to-maximize-prefix-score/rearrange-array-to-maximize-prefix-score.cpp
class Solution {
public:
    int maxScore(vector<int>& nums) {
        long long int result = 0;
        int count = 0;
        sort(nums.begin(), nums.end(), greater<int>());
        for(auto x:nums){
            result += x;
            if(result <= 0)
                break;
            count += 1;
        }
        return count;
    }
};

File: 2655-rearrange-array-to-maximize-prefix-score/rearrange-array-to-maximize-prefix-score.java
class Solution {
    public int maxScore(int[] nums) {
        long result = 0;
        int count = 0;
        Arrays.sort(nums);

        int i = nums.length - 1;
        while(i >= 0 && result + nums[i] > 0) {
            result += nums[i];
            count++;
            i--;
        }

        return count;
    }
}